$OpenBSD: patch-src_rss-config-factory_c,v 1.1 2009/02/13 09:47:58 jasper Exp $

First chunk:
Fix seg. fault when adding new feeds (bz #568857).
From upstream svn -r413.

Second/third chunk:
Fix seg. fault with "realloc(): error: modified chunk-pointer" (bz #562249)
From upstream svn -r444.

--- src/rss-config-factory.c.orig	Fri Feb 13 08:55:44 2009
+++ src/rss-config-factory.c	Fri Feb 13 09:01:14 2009
@@ -279,7 +279,7 @@ create_dialog_add(gchar *text, gchar *feed_text)
         char *gladefile;
   	add_feed *feed = g_new0(add_feed, 1);
 	GladeXML  *gui;
-	gchar *flabel;
+	gchar *flabel = NULL;
   	gboolean fhtml = FALSE;
   	gboolean enabled = TRUE;
   	gboolean del_unread = FALSE;
@@ -521,7 +521,7 @@ feeds_dialog_add(GtkDialog *d, gpointer data)
         add_feed *feed = create_dialog_add(NULL, NULL);
 	if (feed->dialog)
                 gtk_widget_destroy(feed->dialog);
-        GtkWidget *msg_feeds = e_error_new(NULL, "org-gnome-evolution-rss:rssmsg", NULL);
+        GtkWidget *msg_feeds = e_error_new(NULL, "org-gnome-evolution-rss:rssmsg", "", NULL);
 	GtkWidget *progress = gtk_progress_bar_new();
         gtk_box_pack_start(GTK_BOX(((GtkDialog *)msg_feeds)->vbox), progress, FALSE, FALSE, 0);
         gtk_progress_bar_set_fraction((GtkProgressBar *)progress, 0);
@@ -873,7 +873,7 @@ feeds_dialog_edit(GtkDialog *d, gpointer data)
                         add_feed *feed = create_dialog_add(name, feed_name);
                     	if (feed->dialog)
                                 gtk_widget_destroy(feed->dialog);
-        		GtkWidget *msg_feeds = e_error_new(NULL, "org-gnome-evolution-rss:rssmsg", NULL);
+        		GtkWidget *msg_feeds = e_error_new(NULL, "org-gnome-evolution-rss:rssmsg", "", NULL);
 	GtkWidget *progress = gtk_progress_bar_new();
         gtk_box_pack_start(GTK_BOX(((GtkDialog *)msg_feeds)->vbox), progress, FALSE, FALSE, 0);
         gtk_progress_bar_set_fraction((GtkProgressBar *)progress, 0);
